src/main/java/org/alliancegenome/curation_api/dao/ConstructDAO.java

```diff
@@ -0,0 +1,28 @@
+package org.alliancegenome.curation_api.dao;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.persistence.Query;
+
+import org.alliancegenome.curation_api.dao.base.BaseSQLDAO;
+import org.alliancegenome.curation_api.model.entities.Construct;
+
+@ApplicationScoped
+public class ConstructDAO extends BaseSQLDAO<Construct> {
+	
+	@Inject
+	NoteDAO noteDAO;
+
+	protected ConstructDAO() {
+		super(Construct.class);
+	}
+
+	public List<Long> findAllIdsByDataProvider(String dataProvider) {
+		Query jpqlQuery = entityManager.createQuery("SELECT construct.id FROM Construct construct WHERE construct.dataProvider.sourceOrganization.abbreviation = :dataProvider");
+		jpqlQuery.setParameter("dataProvider", dataProvider);
```

Comment:
@markquintontulloch I think we may need to create a method or pass a flag so the find() method so that we get back a list of id's rather than objects. I see keep seeing an increase for this need. Then it prevents us from having to write these methods everywhere.

Reply:
Agreed - will put in another PR for this at some point